dev/NavigationView/NavigationView.cpp

```diff
@@ -1342,9 +1467,46 @@ void NavigationView::OnItemClick(const winrt::IInspectable& /*sender*/, const wi
     if (!m_shouldIgnoreNextSelectionChange && DoesSelectedItemContainContent(clickedItem, itemContainer) && !IsSelectionSuppressed(selectedItem))
     {
         RaiseItemInvoked(selectedItem, false /*isSettings*/, itemContainer);
+        ToggleIsExpandedFromItem(itemContainer);
     }
 }
 
+void NavigationView::ToggleIsExpanded(winrt::NavigationViewItem const& item)
+{
+    if (item)
+    {
+        bool hasChildren = (item.MenuItems().Size() > 0 ||
+                            item.MenuItemsSource() ||
+                            item.HasUnrealizedChildren());
+        if (hasChildren)
+        {
+            auto isItemBeingExpanded = !item.IsExpanded();
+            if (isItemBeingExpanded)
+            {
+                RaiseIsExpanding(item);
+            }
+
+            item.IsExpanded(isItemBeingExpanded);
+
+            if (!isItemBeingExpanded)
+            {
+                RaiseCollapsed(item);
+            }
+        }
+    }
+}
+
+
+void NavigationView::RaiseIsExpanding(winrt::NavigationViewItemBase const& item)
```

Comment:
I didn't see RaiseIsExpanding/RaiseCollapsed implementation, are they just place hold?

Reply:
Yep, they were placeholders, but I hooked them up in the latest commit (it was a small update to do so).